sdk/src/main/java/com/google/cloud/dataflow/sdk/io/DatastoreIO.java

```diff
@@ -888,6 +890,21 @@ public Entity getCurrent() {
     }
 
     @Override
+    public final long getParallelismConsumed() {
+      if (done) {
```

Comment:
I am under the impression that it's important that consumed return 1 when finished, to distinguish it from empty. So given that we have to track doneness, it's not really extra work to return 0 when done in remaining.